pkg/buildermgr/envwatcher.go

```diff
@@ -329,6 +329,14 @@ func (envw *environmentWatcher) createBuilderService(ctx context.Context, env *f
 			Namespace: ns,
 			Name:      name,
 			Labels:    sel,
+			OwnerReferences: []metav1.OwnerReference{
```

Comment:
Ideal way is to use controller ref


https://github.com/kubernetes/apimachinery/blob/37988e577e160bf7749437141fee1d6b8ca15dd6/pkg/apis/meta/v1/controller_ref.go#L54C6-L54C22

We will have to see if we can handle 

```
BlockOwnerDeletion: &blockOwnerDeletion,
		Controller:         &isController,
```

https://github.com/kubernetes-sigs/controller-runtime/blob/8290d13680ed6066d9789c0ed59ff604387e20da/pkg/controller/controller_integration_test.go#L134

Reply:
`BlockOwnerDeletion` and `Controller` set to true should not cause any issue.